databend-jdbc/src/main/java/com/databend/jdbc/DatabendDriverUri.java

```diff
@@ -57,17 +58,22 @@ public final class DatabendDriverUri {
 
     private DatabendDriverUri(String url, Properties driverProperties)
             throws SQLException {
-        Map.Entry<URI, Map<String, String>> uriAndProperties = parse(url);
-        this.properties = mergeProperties(uriAndProperties.getKey(), uriAndProperties.getValue(), driverProperties);
+        Map.Entry<DatabendNodes, Map<String, String>> uriAndProperties = parse(url);
+        List<URI> uris = uriAndProperties.getKey().getUris();
+        this.properties = mergeProperties(uris.get(uris.size() - 1), uriAndProperties.getValue(), driverProperties);
         this.useSecureConnection = SSL.getValue(properties).orElse(false);
         this.useVerify = USE_VERIFY.getValue(properties).orElse(false);
         this.debug = DEBUG.getValue(properties).orElse(false);
         this.strNullAsNull = STRNULL_AS_NULL.getValue(properties).orElse(true);
         this.warehouse = WAREHOUSE.getValue(properties).orElse("");
         this.sslmode = SSL_MODE.getValue(properties).orElse("disable");
         this.tenant = TENANT.getValue(properties).orElse("");
-        this.uri = parseFinalURI(uriAndProperties.getKey(), this.useSecureConnection, this.sslmode);
-        this.address = HostAndPort.fromParts(uri.getHost(), uri.getPort());
+        List<URI> finalUris = parseFinalUris(uris, this.useSecureConnection, this.sslmode);
```

Comment:
what's the expected output of finalUris?

Reply:
oh, this finalUri is not the http query api's final uri

may be we can consider rename it as `processedUris` or `canonicalizeUris`?